plugins/woocommerce/src/Blocks/BlockTypes/Checkout.php

```diff
@@ -450,6 +450,47 @@ protected function enqueue_data( array $attributes = [] ) {
 		$this->asset_data_registry->add( 'collectableMethodIds', $local_pickup_method_ids );
 		$this->asset_data_registry->add( 'shippingMethodsExist', CartCheckoutUtils::shipping_methods_exist() > 0 );
 
+		/**
+		 * This section handles the shipping address fields that trigger shipping rate recalculation.
+		 * The address_fields_for_shipping_rates array contains the default fields that affect shipping rates.
+		 * Additional fields names can be added via the 'woocommerce_essential_shipping_data' filter.
+		 * The filtered array is validated to ensure all entries are strings before being registered.
+		 */
+
+		$address_fields_for_shipping_rates = [ 'state', 'country', 'postcode', 'city' ];
+		/**
+		 * Filter the address fields for shipping rates.
+		 *
+		 * @param array $address_fields The address fields for shipping rates.
+		 * @since 9.9.0
+		 * @return array
+		 */
+		$filtered_fields = apply_filters( 'woocommerce_address_fields_for_shipping_rates', [] );
+
+		if ( is_array( $filtered_fields ) ) {
+			foreach ( $filtered_fields as $key => $value ) {
+				if ( ! is_string( $value ) ) {
+					wc_get_logger()->warning(
+						sprintf(
+							/* translators: %s: address field for shipping rates */
+							__('Address fields for shipping rates values must be strings. Non-string value removed: %s', 'woocommerce' ),
+							$value
+						),
+						array( 'source' => 'woocommerce_address_fields_for_shipping_rates' )
+					);
```

Comment:
```suggestion
					wc_get_logger()->warning(
						sprintf(
							/* translators: %1$s: address field for shipping rates, %2$s index of item in array (number) */
							__( 'Address fields for shipping rates values must be strings. Non-string value removed: %1$s at index %2$s', 'woocommerce' ),
							gettype( $value ),
							$key
						),
						array( 'source' => 'woocommerce_address_fields_for_shipping_rates' )
					);
```

Reply:
I had to change some things for the lint to pass, and failed to notice the failing tests. I'll review you changes, thanks you so much